### PR TITLE
Extend support for analytic boundary conditions

### PR DESCRIPTION
**Pull request by akva2**
- add support for conditions on multiple bases
- add support for analytic robin conditions

also extend the multipatch model generator to generate sets that is used with RT.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/SIM/SIMinput.C [hunk 1]
```diff
@@ -599,13 +599,20 @@ bool SIMinput::parseBCTag (const tinyxml2::XMLElement* elem)
...
-    if (rval)
-      this->setNeumann(rval->Value(),type,direction,code);
-    else
-      this->setNeumann(std::string(),type,direction,code);
+    if (type == "anasol")
+    {
+      IFEM::cout <<" (analytic)";
+      this->setPropertyType(code,Property::ROBIN_ANASOL);
+    }
+    else {
```
Comment (kmokstad) [inline src/SIM/SIMinput.C:607]: Inconsistent style, { after the else here, but on new line i L603.

File: src/SIM/SIMinput.C [hunk 2]
```diff
@@ -599,13 +599,20 @@ bool SIMinput::parseBCTag (const tinyxml2::XMLElement* elem)
...
+      if (!type.empty())
+        IFEM::cout <<" ("<< type <<")";
+      this->setPropertyType(code,Property::ROBIN);
+      if (rval)
+        this->setNeumann(rval->Value(),type,direction,code);
+      else
+        this->setNeumann(std::string(),type,direction,code);
```
Comment (kmokstad) [inline src/SIM/SIMinput.C:614]: or `this->setNeumann(rval ? rval->Value() : "", type, direction,code);`

File: Apps/Common/MultiPatchModelGenerator.C [hunk 3]
```diff
@@ -469,10 +471,19 @@ bool MultiPatchModelGenerator2D::createTopologySets (SIMinput& sim) const
     insertion(e1, e1f, TopItem(i*nx+1,1,1));
     insertion(e2, e2f, TopItem((i+1)*nx,2,1));
   }
+  for (const auto& e : e1)
+    xb.insert(e);
+  for (const auto& e : e2)
+    xb.insert(e);
```
Comment (kmokstad) [inline Apps/Common/MultiPatchModelGenerator.C:477]: All the `e` objects here will be a `TopEntity` reference, so I rather would use `for (const TopEntity& e : e1)`, etc. here, and the reader will save a few seconds figuring out that himself ;)

File: Apps/Common/MultiPatchModelGenerator.C [hunk 4]
```diff
@@ -848,6 +859,22 @@ bool MultiPatchModelGenerator3D::createTopologySets (SIMinput& sim) const
       for (size_t i = 0; i < nx; ++i)
         insertion(TopItem(IJK2K(i,j,k),r,2), "Boundary", "Face");
 
+  TopEntity& bx  = sim.topology("BoundaryX");
+  TopEntity& by  = sim.topology("BoundaryY");
+  TopEntity& bz  = sim.topology("BoundaryZ");
+
+  auto side_insertion = [&sim](TopEntity& set, int start)
```
Comment (kmokstad) [inline Apps/Common/MultiPatchModelGenerator.C:866]: All the other lambda's in this file are declared `auto&&`, so why not here as well for the consistency? (don't remember if there is a practical difference in this case or not).

**Review by kmokstad — APPROVED** [review]
Only a few cosmetic tings

File: src/SIM/SIMinput.C (hunk 2 above)
Comment (akva2) [reply src/SIM/SIMinput.C:614]: sure. not new code but we can improve while at it.

File: Apps/Common/MultiPatchModelGenerator.C (hunk 3 above)
Comment (akva2) [reply Apps/Common/MultiPatchModelGenerator.C:477]: they are Topitem, but sure.

File: Apps/Common/MultiPatchModelGenerator.C (hunk 4 above)
Comment (akva2) [reply Apps/Common/MultiPatchModelGenerator.C:866]: changed.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**